lib/pure/net.nim

```diff
@@ -690,11 +690,10 @@ when defineSsl:
     let ctx = SslContext(context: ssl.SSL_get_SSL_CTX)
     let hintString = if hint == nil: "" else: $hint
     let (identityString, pskString) = (ctx.clientGetPskFunc)(hintString)
-    if psk.len.cuint > max_psk_len:
+    if pskString.len.cuint > max_psk_len:
       return 0
     if identityString.len.cuint >= max_identity_len:
       return 0
-
     copyMem(identity, identityString.cstring, pskString.len + 1) # with the last zero byte
```

Comment:
need to double check this is right, too

Reply:
yes, code in openssl/extensions_cInt:
char identity[PSK_MAX_IDENTITY_LEN + 1];
and tests identity with strlen
but
unsigned char psk[PSK_MAX_PSK_LEN];

and it calls the callback with:
psklen = s->psk_client_callback(s, NULL, identity, sizeof(identity) - 1,
                                        psk, sizeof(psk));
tests that strlen(identity)<PSK_MAX_IDENTITY_LEN
and that psklen<PSK_MAX_PSK_LEN

jesus, lots of trust of client code going on there fucking hell openssl